apps/guardian/src/shared/store.ts

```diff
@@ -156,12 +158,16 @@ export const electionStepAtom = atom<ElectionStep | undefined, [], void>(
             );
             const now = new Date();
 
+            const includedGuardians = guardians.filter((g) => !g.excluded);
             const step = (() => {
-                if (guardian.hasDecryptionShare && guardian.hasDecryptionProof) return TallyStep.Done;
+                if (includedGuardians.every((g) => g.hasDecryptionShare && g.hasDecryptionProof)) return TallyStep.Done;
+                if (guardian.hasDecryptionShare && guardian.hasDecryptionProof) return TallyStep.AwaitPeerShares;
                 if (
-                    guardians.filter(([, g]) => !g.excluded).every(([, g]) => g.hasDecryptionShare) ||
-                    electionConfig.decryptionDeadline < now
+                    electionConfig.decryptionDeadline < now &&
+                    includedGuardians.filter((g) => g.hasDecryptionShare).length < electionConfig.guardianThreshold
                 )
+                    return TallyStep.Incomplete;
+                if (includedGuardians.every((g) => g.hasDecryptionShare) || electionConfig.decryptionDeadline < now)
```

Comment:
Why is there no check for the threshold here? If the deadline has passed but not enough have posted the first share then there is no point in generating responses.